jetbrains-core/src/software/aws/toolkits/jetbrains/core/credentials/AwsConnectionManager.kt

```diff
@@ -195,8 +201,17 @@ abstract class AwsConnectionManager(private val project: Project) : SimpleModifi
      */
     val activeCredentialProvider: ToolkitCredentialsProvider
         @Throws(CredentialProviderNotFoundException::class)
-        get() = selectedCredentialsProvider ?: throw CredentialProviderNotFoundException(message("credentials.profile.not_configured")).also {
-            LOGGER.warn(IllegalStateException()) { "Using activeCredentialProvider when credentials is null, calling code needs to be migrated to handle null" }
+        get() {
+            val state = connectionState
+            return if (state is ConnectionState.ValidConnection) {
+                state.credentials
+            } else {
+                throw CredentialProviderNotFoundException(message("credentials.profile.not_configured")).also {
+                    LOGGER.warn(IllegalStateException()) {
+                        "Using activeCredentialProvider when credentials is null, calling code needs to be migrated to handle null"
```

Comment:
This will log even if state is not null?

Reply:
Fixed